src/Common/ThreadStatus.cpp

```diff
@@ -157,24 +168,11 @@ ThreadStatus::~ThreadStatus()
 {
     flushUntrackedMemory();
 
-    if (thread_group)
-    {
-        ThreadGroupStatus::ProfileEventsCountersAndMemory counters
-        {
-            performance_counters.getPartiallyAtomicSnapshot(),
-            memory_tracker.get(),
-            thread_id
-        };
-
-        std::lock_guard guard(thread_group->mutex);
-        thread_group->finished_threads_counters_memory.emplace_back(std::move(counters));
-        thread_group->threads.erase(this);
```

Comment:
Shouldn't we do unlink here?
(probably it is moved to some other place, just want to double-check)

Reply:
Unlink is done in

if (deleter)
deleter();